Tests/SecureXPCTests/Client & Server/Server Error Handler Test.swift

```diff
@@ -0,0 +1,80 @@
+//
+//  Server Error Handler Test.swift
+//  
+//
+//  Created by Josh Kaplan on 2022-02-23.
+//
+
+import Foundation
+import XCTest
+import SecureXPC
+
+// Note: it's intentional this is *not* `Codable` to validate that server side all errors are sent to the error handler
+private enum ExampleError: Error, Equatable {
+    case completeAndUtterFailure
+}
+
+class ServerErrorHandlerTest: XCTestCase {
+    
+    func testErrorHandler_Sync()  throws {
+        let errorToThrow = ExampleError.completeAndUtterFailure
+        let errorExpectation = self.expectation(description: "\(errorToThrow) should be provided to error handler")
+        
+        let failureRoute = XPCRoute.named("always", "throws")
+        let server = XPCServer.makeAnonymous()
+        let client = XPCClient.forEndpoint(server.endpoint)
+        server.registerRoute(failureRoute) {
+            throw ExampleError.completeAndUtterFailure
+        }
+        server.setErrorHandler { error in
+            switch error {
+                case .handlerError(let error):
+                    if case let .available(underlyingError) = error.underlyingError,
+                       underlyingError as? ExampleError == errorToThrow {
+                        errorExpectation.fulfill()
+                    } else {
+                        XCTFail("Unexpected underlying error: \(error)")
+                    }
+                default:
+                    XCTFail("Unexpected error: \(error)")
+            }
+        }
+        
+        server.start()
+        
+        client.send(route: failureRoute, onCompletion: nil)
+        
+        self.waitForExpectations(timeout: 1)
+    }
+    
+    func testErrorHandler_Async() throws {
+        let errorToThrow = ExampleError.completeAndUtterFailure
+        let errorExpectation = self.expectation(description: "\(errorToThrow) should be provided to error handler")
+        
+        let failureRoute = XPCRoute.named("always", "throws")
+        let server = XPCServer.makeAnonymous()
+        let client = XPCClient.forEndpoint(server.endpoint)
+        server.registerRoute(failureRoute) {
+            throw ExampleError.completeAndUtterFailure
+        }
+        server.setErrorHandler { (error: XPCError) async -> Void in
```

Comment:
Hmmmm I'm wondering if this closure-based error handling API makes sense in an in the new concurrency world.

I see the benefit they have w.r.t. centralizing error handling for common failures, but they can be a bit cumbersome to unwrap. I can't think of anything better, though.

Reply:
On the server side, I don't think the introduction of `async` changes much. Concurrent execution of server handlers was always easily achievable, it's just now possible for them to yield partway through their execution.

However, you bringing this up makes me wonder about few different alternatives: 
1. Each route can optionally register an error handler in addition to the current "operation" handler, which receives `XPCError`s just like the global handler does at present. There's no global handler provided by the framework, but API users can always register the same error handler for all routes.
2. The same as (1), but a global handler can set and receives errors for routes that don't have an error handler set.
3. Each route can optionally register an error handler in addition to the current "operation" handler, which receives `Error`s that it threw during execution. (Potentially also decoding errors for just the payload.) All other errors are sent to a global handler provided by the framework.

I quite like the idea of (3). Thoughts? We can also move this to a discussion.

Reply:
Yeah, worth opening an issue/discussion. I don't currently have any particular thoughts though, because all my server routes do `try!` with a `// TODO: handle errors` :')